parcels/particle.py

```diff
@@ -290,55 +350,58 @@ def execute(self, pyfunc=AdvectionRK4, starttime=None, endtime=None, dt=1.,
+            interval = interval.total_seconds()
         if isinstance(starttime, datetime):
             starttime = (starttime - self.time_origin).total_seconds()
         if isinstance(endtime, datetime):
             endtime = (endtime - self.time_origin).total_seconds()
 
-        # Check if starttime, endtime, runtime and dt are consistent and compute timesteps
+        # Derive starttime, endtime and interval from arguments or grid defaults
         if runtime is not None and endtime is not None:
             raise RuntimeError('Only one of (endtime, runtime) can be specified')
         if starttime is None:
-            if dt > 0:
-                starttime = self.grid.time[0]
-            else:
-                starttime = self.grid.time[-1]
+            starttime = self.grid.time[0] if dt > 0 else self.grid.time[-1]
         if runtime is not None:
-            if dt > 0:
-                endtime = starttime + runtime
-            else:
-                endtime = starttime - runtime
+            endtime = starttime + runtime if dt > 0 else starttime - runtime
         else:
             if endtime is None:
-                if dt > 0:
-                    endtime = self.grid.time[-1]
-                else:
-                    endtime = self.grid.time[0]
-        if endtime < starttime and dt > 0:
-            dt = -1. * dt
-            print("negating dt because running in time-backward mode")
-        if endtime > starttime and dt < 0:
-            dt = -1. * dt
-            print("negating dt because running in time-forward mode")
-        timesteps = int((endtime - starttime) / dt)
-
-        # Check if output is required and compute outer leaps
-        if output_interval <= 0 or output_interval > abs(endtime-starttime):
-            output_steps = timesteps
-        else:
-            output_steps = abs(int(output_interval / dt))
-        timeleaps = int(timesteps / output_steps)
-        # Execute kernel in sub-stepping intervals (leaps)
-        current = starttime
+                endtime = self.grid.time[-1] if dt > 0 else self.grid.time[0]
+        if interval is None:
+            interval = endtime - starttime
+
+        # Ensure that dt and interval have the correct sign
+        if endtime > starttime:  # Time-forward mode
+            if dt < 0:
+                dt *= -1.
+                print("negating dt because running in time-forward mode")
+            if interval < 0:
+                interval *= -1.
+                print("negating interval because running in time-forward mode")
+        if endtime < starttime:  # Time-backward mode
+            if dt > 0.:
+                dt *= -1.
+                print("negating dt because running in time-backward mode")
+            if interval > 0.:
+                interval *= -1.
+                print("negating interval because running in time-backward mode")
+
+        # Initialise particle timestepping
+        for p in self:
+            p.time = starttime
```

Comment:
This has not been possible before either, so this changes does not affect any current features. In fact it is a first step towards a more complex time-stepping run-time that allows different start times. However, such changes are beyond the scope of this PR, and require some further careful design.